### PR TITLE
RetroPlayer: Code improvements

### PR DESCRIPTION
**Pull request by garbear**
## Description

This PR contains the (mostly) non-functional code improvements from the [Savestate rendering PR](https://github.com/xbmc/xbmc/pull/22635).

Breaking this out for easier backporting and because these changes should be quick and easy to review.

## Motivation and context

Reduce the size of the savestate rendering PR.

## How has this been tested?

Test builds uploaded to https://github.com/garbear/xbmc/releases.

Compiled successfully. Runtime tested as part of https://github.com/xbmc/xbmc/pull/22635 and against current master.

## What is the effect on users?

* None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/cores/RetroPlayer/rendering/RPRenderManager.h [hunk 1]
```diff
@@ -205,7 +206,7 @@ class CRPRenderManager : public IRenderManager, public IRenderCallback
   std::set<std::shared_ptr<CRPBaseRenderer>> m_renderers;
   std::vector<IRenderBuffer*> m_pendingBuffers; // Only access from game thread
   std::vector<IRenderBuffer*> m_renderBuffers;
-  std::map<AVPixelFormat, SwsContext*> m_scalers;
+  std::map<AVPixelFormat, std::map<AVPixelFormat, SwsContext*>> m_scalers; // From -> to -> context
```
Comment (lrusak) [inline xbmc/cores/RetroPlayer/rendering/RPRenderManager.h:209]: Might be a nice opportunity for a unique_ptr with a custom deleted for SwsContext.

**Review by lrusak — APPROVED** [review]
A nice improvement.

File: xbmc/cores/RetroPlayer/rendering/RPRenderManager.h (hunk 1 above)
Comment (garbear) [reply xbmc/cores/RetroPlayer/rendering/RPRenderManager.h:209]: I think we have an avalanche of code to review for the savestate manager and I'd rather leave incremental improvements out and keep it to fixes and features.

**Comment by garbear** [comment]
Found another dialog control ID conflict, pushed a fixup that I'll squash before merge.

**Pull request merged**